lib/UserInterface/textinput/Editor.cpp

```diff
@@ -449,6 +450,9 @@ namespace textinput {
         ProcessMove(kMoveEnd, R);
         std::vector<std::string> completions;
         TabCompletion* tc = fContext->GetCompletion();
+        Reader* reader = fContext->GetReaders()[0];
+        StreamReaderUnix* streamReader = (StreamReaderUnix*)(reader);
```

Comment:
What if the reader is a `StreamReaderWindows`? Can we make `IsFromTTY()` a pure virtual function in `Reader` and implement it for `StreamReaderWindows`, too?